src/gate_detect/include/Gate.h

```diff
@@ -0,0 +1,117 @@
+//
+// Created by da-cam on 20/02/18.
+//
+
+#ifndef PROJECT_GATE_H
+#define PROJECT_GATE_H
+
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+class Gate {
+  public:
+    Gate();
+
+    /*
+     * Function to be called on cv::Mat to find vertical poles and horizontal
+     * poles
+     *
+     * mat : cv::Mat potentially containing a gate
+     *
+     */
+
+    std::vector<float> initialize(const cv::Mat matin);
+
+    /*
+     * Function to check if Gate object has been initialized
+     */
+
+    bool checkMat();
+
+    void vertLineCheck();
+
+    void horLineCheck();
+
+  private:
```

Comment:
picky, but please review naming guidelines, these variables are all named incorrectly and also inconsistently. Also please consider if a variable needs to be a permanent member or if they can be temporary variables.

Reply:
Changed many variables to be local instead of global